### PR TITLE
Implement Features

### PR DESCRIPTION
**Pull request by ElbekDeveloper**
The following features are implemented in this pull request: 
1. Search Vendor by name feature 
2. Simplification of vendor model as requested by frontend guys 
3. Initial test for vendor service

### CODE REVIEW

File: ExadelBonusPlus.DataAccess/VendorRepository.cs [hunk 1]
```diff
@@ -15,10 +14,9 @@ public VendorRepository(IOptions<MongoDbSettings> mongoDbSettings) : base(mongoD
         {
 
         }
-        //should implement
-        public Task<IEnumerable<Vendor>> SearchVendorByLocationAsync(string city, CancellationToken cancellationToken)
+        public async Task<List<Vendor>> SearchVendorByNameAsync(string name, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            return await GetCollection().AsQueryable().Where(v=> v.Name.ToLower().Contains(name.ToLower())).ToListAsync(cancellationToken);
```
Comment (myuk96) [inline ExadelBonusPlus.DataAccess/VendorRepository.cs:19]: in which query is it going to serialize?

File: ExadelBonusPlus.Services.Models/MapperProfile.cs [hunk 2]
```diff
@@ -45,8 +45,10 @@ public MapperProfile()
             CreateMap<Location, LocationDto>().ReverseMap();
             CreateMap<AddVendorDto, Vendor>()
                 .AfterMap((src, dest) => dest.Id = Guid.NewGuid());
+            //From dto to model
```
Comment (myuk96) [inline ExadelBonusPlus.Services.Models/MapperProfile.cs:48]: the comment is useless

File: ExadelBonusPlus.WebApi/IServiceCollectionExtensions.cs [hunk 3]
```diff
@@ -39,6 +34,9 @@ public static void AddVendorTransient(this IServiceCollection services)
         {
             services.AddScoped<IVendorService, VendorService>();
             services.AddScoped<IVendorRepository, VendorRepository>();
+
+            services.AddScoped<IValidator<AddVendorDto>, AddVendorDtoValidator>();
```
Comment (myuk96) [inline ExadelBonusPlus.WebApi/IServiceCollectionExtensions.cs:38]: is it going to be different for every request?
why scoped?

**Review by myuk96 — CHANGES_REQUESTED** [review]

File: ExadelBonusPlus.DataAccess/VendorRepository.cs (hunk 1 above)
Comment (ElbekDeveloper) [reply ExadelBonusPlus.DataAccess/VendorRepository.cs:19]: That is temporary implementation of search. It uses linq. This week during the refactoring, I will come back to this part and will be realizing the search via indexes in mongodb

File: ExadelBonusPlus.Services.Models/MapperProfile.cs (hunk 2 above)
Comment (ElbekDeveloper) [reply ExadelBonusPlus.Services.Models/MapperProfile.cs]: Updated

**Pull request merged**